node/node_test.go

```diff
@@ -34,9 +33,9 @@ func TestRun(t *testing.T) {
 	cometListener, err := net.Listen("tcp", cometHTTPAddress)
 	require.NoError(t, err)
 	app := testapp.NewTest(t, chainID.String())
-	ethstatedb := rawdb.NewMemoryDatabase()
+	ethstatedb := testutils.NewEthStateDB(t)
 	defer func() {
-		require.NoError(t, ethstatedb.Close())
+		require.NoError(t, ethstatedb.TrieDB().Close())
```

Comment:
Can we change `testutils.NewEthStateDB` to include

```go
t.Cleanup(func() {
	require.NoError(t, ethstatedb.TrieDB().Close())
})
```
that way we can remove this defer?

Reply:
Could you add this change to `testutils.NewEthStateDB` like @joshklop  suggested so we don't need to include the cleanup steps in the test runner?

Reply:
We'll still want to add this to the test utils. I think you should be able to refactor `testutils/utils.go#NewEthStateDB` like this to support the cleanup in the helper instead of the test itself.

```
func NewEthStateDB(t *testing.T) state.Database {
	rawstatedb := rawdb.NewMemoryDatabase()
	t.Cleanup(func() {
		require.NoError(t, rawstatedb.Close())
	})
	ethstatedb := state.NewDatabase(rawstatedb)
	t.Cleanup(func() {
		require.NoError(t, ethstatedb.TrieDB().Close())
	})
	return ethstatedb
}
```

Reply:
Or even better - include both `t.Cleanup` statements in a single one :)